packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -198,20 +211,43 @@ export class Publisher {
           return batch.resolveEvents()
         }
 
+        let authString = undefined
+        if (this._tokenManager) {
+          const tokenPromise = this._tokenManager.getAccessToken()
+          const token = await tokenPromise
+          if (token && token.access_token) {
+            authString = `Bearer ${token.access_token}`
+          }
+        }
+
+        let headers
+        if (authString) {
+          headers = {
+            'Content-Type': 'application/json',
+            Authorization: authString,
+            'User-Agent': 'analytics-node-next/latest',
+          }
+        } else {
+          headers = {
+            'Content-Type': 'application/json',
+            'User-Agent': 'analytics-node-next/latest',
+          }
+        }
+
         const request: HTTPClientRequest = {
           url: this._url,
           method: 'POST',
-          headers: {
-            'Content-Type': 'application/json',
-            Authorization: `Basic ${this._auth}`,
-            'User-Agent': 'analytics-node-next/latest',
-          },
-          data: { batch: events, sentAt: new Date() },
+          headers: headers,
+          body: JSON.stringify({
+            batch: events,
+            writeKey: this._writeKey,
+            sentAt: new Date(),
+          }),
           httpRequestTimeout: this._httpRequestTimeout,
         }
 
         this._emitter.emit('http_request', {
-          body: request.data,
+          body: request.body,
```

Comment:
I suppose this is also a breaking change (or maybe technically a fix since it was originally a string)